http.go

```diff
@@ -1291,20 +1291,14 @@ func (resp *Response) ReadLimitBody(r *bufio.Reader, maxBodySize int) error {
 
 	if !resp.mustSkipBody() {
 		err = resp.ReadBody(r, maxBodySize)
-		if errors.Is(err, syscall.ECONNRESET) {
-			return nil
-		}
-		if err != nil {
+		if err != nil && !errors.Is(err, syscall.ECONNRESET) {
 			return err
 		}
 	}
 
 	if resp.Header.ContentLength() == -1 {
 		err = resp.Header.ReadTrailer(r)
-		if errors.Is(err, syscall.ECONNRESET) {
-			return nil
-		}
-		if err != nil && err != io.EOF {
+		if err != nil && err != io.EOF && !errors.Is(err, syscall.ECONNRESET) {
```

Comment:
same as above

Reply:
@moredure OK. Changed to your recomendation.